src/functions/fetchRequests/fetchRequestsUser.js

```diff
@@ -0,0 +1,257 @@
+export const deleteUser = async (userObj, setUser) => {
+    try {
+        // DELETE /user {token} => success/failure 
+        const res = await fetch(`${process.env.REACT_APP_REST_API}/user`, {
+            method: "DELETE",
+            headers: { 
+                "Content-Type": "application/json",
+                "authorization": userObj?.authToken,
+            },
+        });
+
+        const data = await res.json();
+        console.log('-> deleteUser(), data:', data);
+
+        if (!res.ok || data.error) throw new Error(`FetchError ${res.status} : ${data.error || data}`)
+
+        // remove the user from state, remove the token
+        setUser( {} );
+
+        // returns the result
+        return res.ok;
+
+    } catch (error) {
+        console.log('-> deleteUser(), error: ',error);
+        return false;
+    }
+
+};
+
+
+// loginUser( {userObj}, setUser ) => fetch(); use setUser( loggedUser ); save token
+export const loginUser = async (userObj, setUser) => {
+    try {
+        console.log('->loginUser() run with userObj: ', userObj)
+        // POST /user/login {username, password} => {userObj, token}
+        const res = await fetch(`${process.env.REACT_APP_REST_API}/user/login`, {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify( userObj ),
+        });
+
+        const data = await res.json();
+        console.log('->loginUser(), data: ',data);
+
+        if (!res.ok || data.error) throw new Error(`FetchError ${res.status} : ${data.error || data}`)
+
+        // save user token 
+        data.user.authToken = data.token
+
+        // set the user to state
+        setUser( data.user );
+
+        // returns the result
+        return res.ok;
+
+    } catch (error) {
+        console.log('-> loginUser(), error: ',error);
+        return false;
+    }
+};
+
+
+// logoutUser( token ) => remove token; setUser( {} )
+export const logoutUser = async (setUser) => {
+
+    setUser( {} );
+
+};
+
+
+// retrieveUser( token ) => fetch(token); retrieve userObj
+export const retrieveUser = async (setUser) => {
+    try {
+        // GET /user/self {token} => {userObj}
+        const res = await fetch(`${process.env.REACT_APP_REST_API}/user/self`, {
+            method: "GET",
+            headers: { 
+                "Content-Type": "application/json", 
+                "authorization": userObj?.authToken,
+            }
+        });
+
+        const data = await res.json();
+        console.log('-> retrieveUser(), data: ',data);
+
+        if (!res.ok || data.error) throw new Error(`FetchError ${res.status} : ${data.error || data}`)
+
+        // set the user to state
+        setUser( data );
+
+    } catch (error) {
+        console.log('-> retrieveUser(), error: ',error);
+        return false;
+    }
+}
+
+
+
+// retrieveUserByUsername( username ) => fetch(token); retrieve userObj
```

Comment:
We don't know how users will be retrieved or in what context. Please reomve to avoid confusion